application/src/main/java/org/thingsboard/server/service/install/SqlDatabaseUpgradeService.java

```diff
@@ -701,6 +710,28 @@ public void upgradeDatabase(String fromVersion) throws Exception {
                         } catch (Exception e) {
                         }
 
+                        PageLink pageLink = new PageLink(1000);
+                        PageData<User> users;
+                        do {
+                            List<ListenableFuture<?>> futures = new ArrayList<>();
+                            users = userService.findAllUsers(pageLink);
+                            for (User user : users.getData()) {
+                                futures.add(dbUpgradeExecutor.submit(() -> {
+                                    JsonNode additionalInfo = user.getAdditionalInfo();
+                                    if (additionalInfo.isObject() && additionalInfo.has("userPasswordHistory")){
```

Comment:
check that additionalInfo is not null.

Reply:
If history is empty, we don't need to update the credentials, right?

Reply:
yes